internal/controllers/fake/client.go

```diff
@@ -81,3 +93,24 @@ func (f fakeStatusWriter) Patch(ctx context.Context, obj ctrlclient.Object, patc
 	}
 	return json.Unmarshal(content, obj)
 }
+
+// simulateMarshalRoundtrip makes API types behave closer to reality for the
+// fake client.
+//
+// NOTE(nick): We've had a lot of bugs due to the way the apiserver
+// modifies and truncates objects on update. We simulate this behavior
+// to catch this class of bug.
+//
+// The `apiResp` argument is marshaled and then unmarshalled into `dest` arg.
+// (This allows the fake client to mutate *copy* of the user-provided obj, and
```

Comment:
(tiny nit) should this be:
```suggestion
// (This allows the fake client to mutate a *copy* of the user-provided obj, and
```
or maybe:
```suggestion
// (This allows the fake client to mutate the *copy* of the user-provided obj, and
```
?